packages/telemetry/src/slog-to-otel.js

```diff
@@ -347,110 +407,167 @@ export const makeSlogToOtelKit = (tracer, overrideAttrs = {}) => {
-      spans.push('init');
-      initing = false;
-    }
-
     if (finished) {
       return;
     }
 
     // console.log('slogging', obj);
     switch (slogType) {
+      case 'kernel-init-start': {
+        spans.push('init');
+        break;
+      }
+      case 'kernel-init-finish': {
+        spans.pop('init');
+        break;
+      }
+      case 'bundle-kernel-start': {
+        assert.equal(spans.topKind(), 'init');
+        spans.push('bundle-kernel');
+        break;
+      }
+      case 'bundle-kernel-finish': {
+        spans.pop('bundle-kernel');
+        break;
+      }
       case 'import-kernel-start': {
-        spans.start('import-kernel', spans.top());
+        assert.equal(spans.topKind(), 'init');
+        spans.push('import-kernel');
         break;
       }
       case 'import-kernel-finish': {
-        spans.end('import-kernel');
-        spans.pop('init');
+        spans.pop('import-kernel');
         break;
       }
       case 'create-vat': {
         const { vatSourceBundle: _2, name, ...attrs } = slogAttrs;
-        spans.start(['create-vat', slogAttrs.vatID], spans.top(), {
-          'vat.name': name,
-          ...attrs,
+        spans.push(['create-vat', slogAttrs.vatID], {
+          attributes: {
+            'vat.name': name,
+            ...attrs,
+          },
         });
-        spans.end(['create-vat', slogAttrs.vatID]);
         vatIdToAttrs.init(slogAttrs.vatID, { name, ...attrs });
         break;
       }
       case 'vat-startup-start': {
-        spans.start(['vat-startup', slogAttrs.vatID], spans.top());
+        spans.push(['vat-startup', slogAttrs.vatID]);
         break;
       }
       case 'vat-startup-finish': {
-        spans.end(['vat-startup', slogAttrs.vatID]);
+        spans.pop(['vat-startup', slogAttrs.vatID]);
+        spans.pop(['create-vat', slogAttrs.vatID]);
         break;
       }
       case 'start-replay': {
-        const ckey = getCrankKey(true);
-        // Replay may not happen in the context of a crank.
-        replayKey = spans.has(ckey) && ['start-replay', slogAttrs.vatID];
-        spans.startNamed(
-          `replay ${slogAttrs.vatID}`,
-          replayKey || ckey,
-          spans.top(),
-          slogAttrs,
-        );
+        assert(!isReplaying);
+        spans.push(['vat-replay', slogAttrs.vatID]);
+        isReplaying = true;
         break;
       }
       case 'finish-replay': {
-        spans.end(replayKey || getCrankKey());
-        replayKey = undefined;
+        spans.pop(['vat-replay', slogAttrs.vatID]);
+        assert(isReplaying);
+        isReplaying = false;
         break;
       }
       case 'deliver': {
         const {
           vd: [delivery],
           kd,
+          _replay,
           ...attrs
         } = slogAttrs;
+        if (isReplaying) {
+          spans
+            .top()
+            ?.addEvent(
+              `deliver-${delivery}`,
+              cleanAttrs({ delivery, ...attrs }),
+              now,
+            );
```

Comment:
Empirically, I've verified that it is.  However, it would be easy enough to check.  My only concern is that it makes things a bit more complicated and adds a failure mode -- I'm a subscriber to the principle of not checking for error conditions that you're not prepared to deal with anyway.